### ISSUE TITLE
Utterance queue vs. aria value text

### ISSUE THREAD

**Issue by twant** [phetsims/molarity#82]
@emily-phet, @Matthew-phet, @jessegreenberg, @zepumph and I were discussing the implementation of the alerts in Molarity. Originally, all of the alert text was being read out via utterance queue, but @zepumph and I changed it so that most of the slider alert text is now read out via aria value text (including the on focus alerts, which differ from the alerts read out when the sliders are moved). We think that it sounds the way that it should and avoids duplicating the alert text (which was happening when we had some of the alert text as aria-valueText and some in the utterance queue), particularly when the "show values" checkbox is checked, but we wanted to check with you @terracoda and see what you thought about an approach that relies mostly on aria-valueText instead of utterance queue. Are there any major downsides that we should know about as we pursue this implementation approach?

**Comment by zepumph** [comment]
In addition. @terracoda the way that the "on focus aria-valuetext" api is implemented requires there to be "on value change" valuetext accompanying it. I think it would have been equally possible to put both communications (on focus and on value change) into utterance alerts instead of using aria-valuetext. The issue was when we were trying to use on focus valuetext and on value change alerts.

**Comment by terracoda** [comment]
Hmm, this an interesting issue.
For me, it is not just delivering the correct strings. I agree that Molarity text sound rather good. 

The way I think about aria-valuetext vs. aria-live alerts is that aria-value text only contains information directly about the object the slide/spinbutton is controlling.

The aria-live text would bring in related information about what is happening.

I think keeping the descriptions separate like that makes the system more modular and potentially more robust. I will gather some examples.

**Comment by terracoda** [comment]
Hmm, this an interesting issue.
For me, it is not just delivering the correct strings. I agree that Molarity text sound rather good. 

The way I think about aria-valuetext vs. aria-live alerts is that aria-value text only contains information directly about the object the slide/spinbutton is controlling.

The aria-live text would bring in related information about what is happening.

I think keeping the descriptions separate like that makes the system more modular and potentially more robust. I will gather some examples.

**Comment by Matthew-phet** [comment]
@twant So going forwards its been decided to change where some of the alerts go

**Solute Slider Changes Unsaturated**
This was what it looked like when everything was in aria value text.
OLD: More solute, more concentrated. A low amount of ‪Drink mix‬. Solution not concentrated.

This is what we want it to look like now.
1. Interactive alert: More/Less solute, more/less concentrated. Solution not concentrated
2. Aria value text: {{A low amount}} of ‪Drink mix‬. 
          1. For the aria value text it should only be read on region change

**Solute Slider Changes Saturated**
This was what it looked like when everything was in aria value text.
OLD: More solute, more solids. Still saturated with some solids.

This is what it should look like now.
1. Interactive alert: More/Less solute, more/less solids. Still saturated with {{some}} solids.
2. Aria value text: {{A low amount}} of ‪drink mix
          1. For the aria value text it should only be read on region change


**Solution Volume Slider Changes Unsaturated**
This was what it looked like when everything was in aria value text.
OLD: More solution, less concentrated. Beaker {{half full}}. Solution not concentrated. 

This is what it should look like now.
1. Interactive alert: More/Less solute, more/less concentrated. Solution not concentrated
2. Aria value text: Beaker {{half full}}
          1. For the aria value text it should only be read on region change

**Solution Volume Slider Changes Saturated**
This was what it looked like when everything was in aria value text.
OLD: More solution, less solids. Still saturated with a couple of solids.

This is what it should look like now.
1. Interactive alert: More/Less solution, more/less solids. Still saturated with {{a couple of}} solids.
2. Aria value text: Beaker {{half full}}
          1. For the aria value text it should only be read on region change

**Comment by emily-phet** [comment]
@Matthew-phet 
I think you're missing a "." after "concentrated" in the NEW interactive alerts above. You might edit you're original post, adding those in.

**Comment by twant** [comment]
@Matthew-phet and I listened to this today. So far, the aria valueText seems to be working as designed, but the alerts need to be modified so that "less solute, less concentrated" gets read out every time, and the state info is read out on a region change.

**Comment by twant** [comment]
Also, need to add back onfocus value text.

**Comment by twant** [comment]
It seems that the on-focus value text is now the same as the value text after a slider has been focused and moved. Is this correct @Matthew-phet?

**Comment by twant** [comment]
@Matthew-phet @emily-phet Assigning these back over to you for review.

**Comment by twant** [comment]
@Matthew-phet @emily-phet Question regarding the aria-valueText vs. aria-live alerts changes. Will you be wanting some similar split between aria-valueText and aria-live alerts for quantitative alerts (when the show values checkbox is checked)?

**Comment by Matthew-phet** [comment]
Yes, the split should be the same or similar.

**Comment by twant** [comment]
Great! I took a pass at it, let's review this evening and see how it sounds! @Matthew-phet
